### PR TITLE
make setHidden use forgetTarget

### PR DESCRIPTION
**Pull request by commy2**
**When merged this pull request will:**
- title
Question is if this breaks anything else? Maybe just apply it to enemy groups?

### CODE REVIEW

**Comment by PabstMirror** [comment]
What is locality on forgetTarget?

**Comment by commy2** [comment]
I just assumed it's AG EL, since it's the anti-`reveal`. This effect is a global one.

**Comment by SilentSpike** [comment]
I have a feeling it may be global since it was added in A3

**Comment by PabstMirror** [comment]
Should we `revel` to previous knowsAbout when they wake up?

**Comment by commy2** [comment]
They'd be revealed over time. I think it's better not to.

File: addons/common/XEH_postInit.sqf [hunk 1]
```diff
@@ -54,7 +54,9 @@
             _object setVariable [QGVAR(oldVisibility), _vis];
             _object setUnitTrait ["camouflageCoef", 0];
             {
-                _x forgetTarget _object;
+                if (side _x != side group _object) then {
```
Comment (Dystopian) [inline addons/common/XEH_postInit.sqf]: this will not work if `side _object == ENEMY`

Comment (SilentSpike) [reply addons/common/XEH_postInit.sqf]: We actually just discussed this internally, it's quite edge case. One fix would just be to do:
```
if (side _x != side _object)
```

Comment (SilentSpike) [reply addons/common/XEH_postInit.sqf]: The other would be to just have friendlies forget the unit too (with the exception of their own group for healing).

**Pull request merged**